pkg/utils/template.go

```diff
@@ -0,0 +1,77 @@
+/*
+Copyright 2025 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package utils
+
+import (
+	"bytes"
+    "fmt"
+	"strings"
+    "text/template"
+
+	"github.com/Masterminds/sprig/v3"
+)
+
+type TemplateHelper struct {
+	template *template.Template
+}
+
+func NewTemplateHelper(name string) (*TemplateHelper, error) {
+	tmpl := template.New("").Funcs(sprig.FuncMap()).Option("missingkey=error")
+    tmpl = tmpl.Funcs(template.FuncMap{
+        "asLabelValue": func(s string) string {
+            // Replace spaces and periods with dashes
+            return strings.ReplaceAll(strings.ReplaceAll(s, " ", "-"), ".", "-")
```

Comment:
I'd suggest to add this in a separate PR so we can polish it there. The func needs to be more complex  as
>  label must be an empty string or consist of alphanumeric characters, `-`, `_` or `.`, and must start and end with an alphanumeric character
> A label key and value must begin with a letter or number, and may contain letters, numbers, hyphens, dots, and underscores, up to 63 characters each.

So we'd need something like this:
```golang
	re := regexp.MustCompile(`[^a-zA-Z0-9._-]+`)
	return re.ReplaceAllString(input, "_")
```

And probably even more complex than that so that it would guarantee that the end result is accepted as a label value. Do length checking (trimming) and somehow handle cases where the value would start with non-alphanumeric.

WDYT?

Reply:
I'm not familiar enough with Go to write this correctly in the time that I have. I'm happy to remove this and let you add a better one later